src/bin/cargo/commands/login.rs

```diff
@@ -36,7 +37,7 @@ pub fn cli() -> Command {
 pub fn exec(config: &mut Config, args: &ArgMatches) -> CliResult {
     ops::registry_login(
         config,
-        args.get_one("token").map(String::as_str),
+        args.get_one("token").map(String::as_str).map(Secret::from),
```

Comment:
Maybe we should have a from &str, so that we can remove the repetition hear. What do you think?
(same for the other `.map(Secret::from)`)

Reply:
Hmm, yes good point. Since `Secret<T>` impls `From<T>`, we can do `args.get_one::<String>("token").map(|s| s.as_str().into())`, which might be a little cleaner (that's what I did in the new commit). Or I can add a `fn from_str(&String) -> Secret<&str>` method if you think that'd be nicer :smile:

Reply:
I was thinking of:

```
impl<T, U: From<T>> From<T> for Secret<U> {
    fn from(inner: T) -> Self {
        Self { inner.into() }
    }
}
```
but apparently that does not work.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=b026fc5b25592691eb8fd44f4dd98452

Reply:
Oh sorry, I see now. That would be nice, too bad it doesn't work :disappointed:

Reply:
One option could be to remove the blanket `From<T>` impl and instead implement it individually for only the inner types that we actually wrap with `Secret`, which I think are currently only `String`, `&str`, and `AsymmetricKeyPair`. Then we can do `impl From<&String> for Secret<&str>`, etc.

(Sorry, I think I maybe got a little too excited and made things overly generic...)